packages/cli/src/commands/BaseCommand.ts

```diff
@@ -27,7 +27,7 @@ import { ShutdownService } from '@/shutdown/Shutdown.service';
 export abstract class BaseCommand extends Command {
 	protected logger = Container.get(Logger);
 
-	protected externalHooks?: IExternalHooksClass;
+	protected externalHooks?: ExternalHooks;
```

Comment:
Odd that this needs to be optional.

Reply:
we need to create a new base class maybe for all the servers, instead of making `externalHooks` and `server` optional, or we need to start saving these on the DI container, instead of holding (and passing around) references like these.